### ISSUE TITLE
`CKV_AWS_300`: False positive when S3 lifecycle rule uses intentionally empty prefix

### ISSUE THREAD

**Issue by AX2J-Bixal** [bridgecrewio/checkov#7268]
**Describe the issue**  
We’re seeing a false positive for Checkov rule `CKV_AWS_300` in a valid Terraform S3 lifecycle rule configuration. Specifically, when `abort_incomplete_multipart_upload` is enabled and the lifecycle rule filter explicitly sets an empty prefix (`prefix = ""`) to apply the rule to all objects in the bucket, Checkov flags this as a violation.

The logic behind the change is that an explicit empty prefix is a valid and intentional way to apply lifecycle rules bucket-wide. Checkov currently treats it as if no meaningful filter exists, which causes a false positive.

---

**Examples**  
Example Terraform lifecycle rule triggering the false positive:

```hcl
lifecycle_rule {
  id      = "default"
  enabled = true

  abort_incomplete_multipart_upload {
    days_after_initiation = 7
  }

  filter {
    prefix = "" # intentionally applying to all objects
  }
}
```

**Expected outcome:**  
Checkov should **not** flag this configuration as a `CKV_AWS_300` violation, recognizing the explicit empty prefix as a valid filter.

---

**Version (please complete the following information):**  
- Checkov Version: 3.2.458

---

**Additional context**  
The relevant Checkov code is here:  
https://github.com/bridgecrewio/checkov/blob/main/checkov/terraform/checks/resource/aws/S3AbortIncompleteUploads.py

Currently, users must suppress this false positive with a skip comment, which reduces overall scan clarity and forces bypass of the check in intentional cases. Adjusting the rule to differentiate between a missing filter and an explicit empty prefix would improve accuracy.
